lib/tilelivecopy.js

```diff
@@ -1,25 +1,35 @@
 var url = require('url');
 var tilelive = require('tilelive');
+var tileliveOmivore = require('tilelive-omnivore');
 var TileStatStream = require('tile-stat-stream');
 var queue = require('queue-async');
 var combiner = require('stream-combiner');
 var MigrationStream = require('./migration-stream');
 
 function tilelivecopy(srcUri, s3url, options, callback) {
+  var src_loader = tilelive.load;
   var scheme = 'scanline';
   var protocol = url.parse(srcUri).protocol;
   if (protocol === 'mbtiles:') scheme = 'list';
   if (protocol === 'omnivore:') scheme = 'pyramid';
   options.type = scheme;
+  if (options.bundle === true) {
+    src_loader = function(bundle_uri, cb) {
```

Comment:
My point is that this looks like extraneous code. That is, I don't understand what rabbit hole you're seeing. `tilelive.load` should "just work" with the bundle URI -- it should select the tilelive-omnivore constructor because the URI starts with `omnivore:`.